File: src/easy/repeated-substring-pattern.ts
export const repeatedSubstringPattern = (s: string): boolean => {
	const n = s.length;
	if(n < 2) return false;

	for(let len = 1; len <= n / 2; len++) {
		if( n % len !== 0) continue;
		let valid = true
		for(let i = len; i < n; i++)  {
			if(s[i] !== s[i % len])  {
				valid = false;
				break;
			}
		}
		if(valid) return true
	}
	return false;
}

File: src/easy/repeated_substring_pattern.zig
const std = @import("std");

const Allocator = std.mem.Allocator;

pub fn repeatedSubstringPattern(allocator: Allocator, s: []const u8) ?bool {
    var slices = [_][]const u8{ s, s };
    const result = std.mem.concat(allocator, u8, &slices) catch return null;
    defer allocator.free(result);
    return std.mem.containsAtLeast(u8, result[1 .. s.len * 2 - 1], 1, s);
}
